corehq/util/metrics/metrics.py

```diff
@@ -30,14 +31,19 @@ def _validate_tag_names(tag_names):
 
 
 class MetricBase:
-    def __init__(self, name: str, documentation: str, tag_names: Iterable=tuple(), tag_values=None):
+    def __init__(self, name: str, documentation: str, tag_names: Iterable=tuple(), **kwargs):
         self.name = name
         if not METRIC_NAME_RE.match(name):
             raise ValueError('Invalid metric name: ' + name)
         _enforce_prefix(name, 'commcare')
         self.documentation = documentation
         self.tag_names = _validate_tag_names(tag_names)
-        self.tag_values = tag_values
+        self.tag_values = kwargs.pop('tag_values', None)
+        self._kwargs = kwargs
```

Comment:
Will an error be raised if unknown kwargs (due to typo, etc.) are passed in here? It looks like the answer is "No." Would be great if that situation could be improved to make it more typo-resistant.

Reply:
I agree, this was another part I wasn't very happy with. I struggled to find a clean way to do this. It stems from the fact that the Datadog and Prometheus classes require different arguments. One good thing here is that these classes are never instantiated outside of the package. The public API has all the available kwargs clearly listed.

I think this stems from trying to emulate the Prometheus API: `metric.label(...).inc()` which requires classes.